api/src/main/java/org/openmrs/validator/PersonValidator.java

```diff
@@ -69,23 +69,58 @@ public void validate(Object target, Errors errors) {
 		
 		Person person = (Person) target;
 		
+		validatePersonNames(person, errors);
+		validatePersonAddresses(person, errors);
+		
+		if (!person.getVoided() && !validatePersonHasAtLeastOneNonVoidedName(person)) {
```

Comment:
Did you intentionally change the order of the validation of names and addresses?

Reply:
In Line 74 personName is validated. in Line 89 personAdress in the original code. I am not sure what you mean with change of order?

Reply:
You are validation person addresses on line 73 before the names from line 75

Reply:
Reason was that for me the code is more readable if both method calls followed each other.

Reply:
Changing the order may result into unintended side effects from the caller. So i would be cautious not to do that in this pull request. Am sure you have ever seen bug or unexpected behavior introduced by a mere change of the order for a few lines of code. So let us be as surgical as possible. 😊

Reply:
I reverted the order